src/main/java/com/algorand/algosdk/transaction/Transaction.java

```diff
@@ -745,6 +745,12 @@ public static Transaction createAssetTransferTransaction(// AssetTransaction
         return tx;
     }
 
+    private void setNote(byte[] note) {
+        if (note != null && note.length != 0) {
```

Comment:
What is a valid empty note field? How can one reset the note field to empty?

Reply:
In order for the note field to be omitted, it needs to be `null` in the Transaction object. We were previously checking for null, but allowing it to be an array of length zero.

Reply:
You shouldn't reset the note field, which is why I made this method private. Unfortunately all of the fields are public so it isn't strictly immutable.

Other than the groupID and the fee, I would prefer that the transaction object be immutable.